universityofadelaide/shepherd
# Feature/webops 276 configure crons for shepherd

Add openshift configuration to setup cronjobs for shepherd. 

# Manual testing instructions: 
- spin up minishift 
- follow the updated install instructions. Including getting the openshift docker registry ip
- edit the build config and change the git ref  from develop to the name of this feature branch : 
```
 source:
    git:
      ref: feature/webops-276-configure-crons-for-shepherd
      uri: 'https://github.com/universityofadelaide/shepherd.git'
```
- deploy shepherd on openshift 
- deploy the `shepherd-openshift-cronjob.yml` Once shepherd has been configured and installed.
```bash
oc process -f ./shepherd-openshift-cronjob.yml -p DATABASE_PASSWORD={db-password-here} -p SHEPHERD_WEB_IMAGESTREAM=172.30.1.1:5000/shepherd/shepherd-web-is:latest | oc create -f -
```
- once builds are complete the job should run every minute
- monitor by `oc get jobs --watch` 
- create an example project in shepherd and watch the operation get processed in the job.
